editor/store/effects.js

```diff
@@ -102,41 +100,66 @@ export default {
 			content: getEditedPostContent( state ),
 			id: post.id,
 		};
-
-		dispatch( {
-			type: 'UPDATE_POST',
-			edits: toSend,
-			optimist: { type: BEGIN, id: POST_UPDATE_TRANSACTION_ID },
-		} );
-		dispatch( removeNotice( SAVE_POST_NOTICE_ID ) );
 		const basePath = wp.api.getPostTypeRoute( getCurrentPostType( state ) );
-		wp.apiRequest( { path: `/wp/v2/${ basePath }/${ post.id }`, method: 'PUT', data: toSend } ).then(
-			( newPost ) => {
-				dispatch( resetPost( newPost ) );
-				dispatch( {
-					type: 'REQUEST_POST_UPDATE_SUCCESS',
-					previousPost: post,
-					post: newPost,
-					edits: toSend,
-					optimist: { type: COMMIT, id: POST_UPDATE_TRANSACTION_ID },
-				} );
-			},
-			( err ) => {
-				dispatch( {
-					type: 'REQUEST_POST_UPDATE_FAILURE',
-					error: get( err, [ 'responseJSON' ], {
-						code: 'unknown_error',
-						message: __( 'An unknown error occurred.' ),
-					} ),
-					post,
-					edits,
-					optimist: { type: REVERT, id: POST_UPDATE_TRANSACTION_ID },
+		const isAutosave = action.options && action.options.autosave;
+
+		if ( isAutosave ) {
+			toSend.parent = post.id;
+			wp.apiRequest( { path: `/wp/v2/${ basePath }/${ post.id }/autosaves`, method: 'POST', data: toSend } ).then(
+				( autosave ) => {
+					dispatch( {
+						type: 'RESET_AUTOSAVE',
+						post: autosave,
+					} );
+
+					dispatch( {
+						type: 'REQUEST_POST_UPDATE_SUCCESS',
```

Comment:
Why do we dispatch `REQUEST_POST_UPDATE_SUCCESS` for autosaves? It seems the action is used for two purposes:

- Displaying a notice, which we've explicitly opted out of for autosaves
- Resetting `isSaving` state, which appears to only be considered for full saves, not autosaves

Reply:
> As I recall, Redux optimist behaves in a way that an action dispatched can either be committed (confirmed) or reverted,

Right, I think its mostly useful where you want to show the UI in the updated state "optimistically" (eg adding a todo, you show the todo added), then if things fail for some reason you can show an error and roll back the change. Not sure how useful that is for the autosave?

Reply:
> Why do we dispatch REQUEST_POST_UPDATE_SUCCESS for autosaves? It seems the action is used for two purposes:

good question, let me dig in a bit further to see why this is here

Reply:
>Right, I think its mostly useful where you want to show the UI in the updated state "optimistically" (eg adding a todo, you show the todo added), then if things fail for some reason you can show an error and roll back the change. Not sure how useful that is for the autosave?

I think of it more in the pure data sense: We assume some success state where a new autosave will become the referenced value, and if a failure is to occur, we revert back to the prior state value.

Reply:
@aduth this is used in the `saving` reducer to reset the saving state. This call resets the state, and that gets reflects in the `PostSavedState` component - without this the saving action never completes.

I see how the naming is confusing, maybe I can try resetting `saving` on `RESET_AUTOSAVE` instead.

Reply:
@aduth In https://github.com/WordPress/gutenberg/pull/6257/commits/253110292d9e56776c003d4396a2e5c1d0b49c8b I switched this to resetting saving on the `RESET_AUTOSAVE`, eliminating the need to dispatch REQUEST_POST_UPDATE_SUCCESS